File: leetcode/125-valid-palindrome/ValidPalindrome.java
class Solution {
    private boolean isLetter(char ch) {
        return (ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z') || (ch >= '0' && ch <= '9');
    }
    public boolean isPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;
        
        while (left < right) {
            while(left < right && !isLetter(s.charAt(left))) {
                left++;
            }
            
            while(left < right && !isLetter(s.charAt(right))) {
                right--;
            }
            
            if (Character.toLowerCase(s.charAt(left)) != Character.toLowerCase(s.charAt(right))) {
                return false;
            }
            
            left++;
            right--;
        }
        
        return true;
    }
}

File: leetcode/125-valid-palindrome/valid-palindrome.js

const isPalindrome = s => {
    const re = /[A-Za-z0-9]/;
    
    let i = 0;
    
    let j = s.length - 1;
    
    while(i < j) {
        if (!re.test(s[i])) {
            i++;
        } else if (!re.test(s[j])) {
            j--;
        } else if (s[i++].toLowerCase() !== s[j--].toLowerCase()) {
            return false;
        }
    }
    
    return true;
}
